downstream/modules/terraform-aap/proc-terraform-provider-configuring.adoc

```diff
@@ -0,0 +1,116 @@
+
+= Configuring the provider
+
+[role="_abstract"]
+
+You must configure the provider to allow {Terraform} to reference and manage a subset of {PlatformNameShort} resources.
+
+The provider configuration belongs in the root module of a Terraform configuration. Child modules receive their provider configurations from the root module.
+
+.Prerequisites
+
+* You have installed and configured link:https://developer.hashicorp.com/terraform/enterprise[Terraform Enterprise or HCP Terraform].
+* You have installed the latest version of link:https://registry.terraform.io/providers/ansible/aap/latest[`terraform-provider-aap`] from the Terraform registry.
+* You have created a username and password.
+
+.Procedure
+
+. Create an inventory resource file. Include a `provider` block. The name given in the block header is the link:https://developer.hashicorp.com/terraform/language/providers/requirements#local-names[local name] of the provider to configure. This provider should already be included in a link:https://registry.terraform.io/providers/ansible/aap/latest/docs[`required_providers` block].
++
+**Example**
++
+----
+# This example creates an inventory named `My new inventory`
+# and adds a host `tf_host` and a group `tf_group` to it,
+# and then launches a job based on the "Demo Job Template" 
+# in the "Default" organization using the inventory created.
+#
+terraform {
+  required_providers {
+    aap = {
+      source = "ansible/aap"
+    }
+  }
+}
+
+provider "aap" {
+  host     = "https://AAP_HOST"
+  username = "ansible"
+  password = "test123!"
+}
+
+resource "aap_inventory" "my_inventory" {
+  name         = "My new inventory"
+  description  = "A new inventory for testing"
+  organization = 1
+  variables = jsonencode(
+    {
+      "foo" : "bar"
+    }
+  )
+}
+
+resource "aap_group" "my_group" {
+  inventory_id = aap_inventory.my_inventory.id
+  name         = "tf_group"
+  variables = jsonencode(
+    {
+      "foo" : "bar"
+    }
+  )
+}
+
+resource "aap_host" "my_host" {
+  inventory_id = aap_inventory.my_inventory.id
+  name         = "tf_host"
+  variables = jsonencode(
+    {
+      "foo" : "bar"
+    }
+  )
+  groups = [aap_group.my_group.id]
+}
+
+data "aap_job_template" "demo_job_template" {
+  name              = "Demo Job Template"
+  organization_name = "Default"
+}
+
+  # In order for passing the inventory id to the job template execution, the Inventory on the job template needs to be set to "prompt on launch"
+resource "aap_job" "my_job" {
+  inventory_id    = aap_inventory.my_inventory.id
+  job_template_id = aap_job_template.demo_job_template.id
+
+  # This resource creation needs to wait for the host and group to be created in the inventory
+  depends_on = [
+    aap_host.my_host,
+    aap_group.my_group
+  ]
+}
+----
+
+. Add the configuration arguments, as shown in the previous example. Optional schema includes:
+
+* **`host`:** (String) AAP server URL. Can also be configured using the AAP_HOSTNAME environment variable.
```

Comment:
```suggestion
* **`host`:** (String) AAP server URL. Can also be configured using the AAP_HOST environment variable.
```

In 1.3.0 and prior, the env variable for this was actually `AAP_HOST`, but it's not [documented](https://registry.terraform.io/providers/ansible/aap/1.3.0/docs). 1.3.0 won't read `AAP_HOSTNAME`, so if we want to be consistent with 1.3.0 this should be `AAP_HOST`.

For 1.4.0 and later we [support (and prefer) `AAP_HOSTNAME`](https://registry.terraform.io/providers/ansible/aap/latest/docs#optional). `AAP_HOSTNAME` aligns with other Ansible content. Since we started with `AAP_HOST`, 1.4.0 and later will fall back to use `AAP_HOST` for compatibility if `AAP_HOSTNAME` is not set.

Reply:
Thanks. Fixed and will change for the 1.4.0 content later.